Sources/API/Routes/PlanRoutes.swift

```diff
@@ -0,0 +1,170 @@
+     
+     - parameter trialPeriodDays:       Specifies a trial period in (an integer number of) days.
+     
+     - returns: A StripeRequest<> item which you can then use to convert to the corresponding node.
+     */
+    
+    public func create(id: String, amount: Int, currency: StripeCurrency, interval: StripeInterval, name: String, intervalCount: Int?, metadata: Node?, statementDescriptor: String?, trialPeriodDays: Int?) throws -> StripeRequest<Coupon> {
+        var body = Node([:])
+        
+        body["id"] = Node(id)
+        
+        body["amount"] = Node(amount)
+        
+        body["currency"] = Node(currency.rawValue)
+        
+        body["interval"] = Node(interval.rawValue)
+        
+        body["name"] = Node(name)
+        
+        if let intervalCount = intervalCount {
+            body["interval_count"] = Node(intervalCount)
+        }
+        
+        if let metadata = metadata?.object {
+            for (key, value) in metadata {
+                body["metadata[\(key)]"] = value
+            }
+        }
+        
+        if let statementDescriptor = statementDescriptor {
+            body["statement_descriptor"] = Node(statementDescriptor)
+        }
+        
+        if let trialPeriodDays = trialPeriodDays {
+            body["trial_period_days"] = Node(trialPeriodDays)
+        }
+        
+        return try StripeRequest(client: self.client, method: .post, route: .plans, query: [:], body: Body.data(body.formURLEncoded()), headers: nil)
+    }
+    
+    /**
+     Retrieve a plan
+     Retrieves the plan with the given ID.
+     
+     - parameter planId: The ID of the desired plan.
+     
+     - returns: A StripeRequest<> item which you can then use to convert to the corresponding node
+     */
+    
+    public func retrieve(plan planId: String) throws -> StripeRequest<Plan> {
+        return try StripeRequest(client: self.client, method: .get, route: .plan(planId), query: [:], body: nil, headers: nil)
+    }
+    
+    /**
+     Update a plan
+     Updates the name or other attributes of a plan. Other plan details (price, interval, etc.) are, by design, not editable.
+     
+     - parameter metaData:              A set of key/value pairs that you can attach to a plan object.
+     
+     - parameter name:                  Name of the plan, to be displayed on invoices and in the web interface.
+     
+     - parameter statementDescriptor:   An arbitrary string to be displayed on your customer’s credit card statement.
+     
+     - parameter trialPeriodDays:       Specifies a trial period in (an integer number of) days.
+     
+     - parameter planId:                The identifier of the plan to be updated.
+     
+     - returns: A StripeRequest<> item which you can then use to convert to the corresponding node
+     */
+    
+    public func update(metadata: Node?, name: String?, statementDescriptor: String?, trialPeriodDays: Int?, forPlanId planId: String) throws -> StripeRequest<Plan> {
```

Comment:
Again my thinking was to let people know and to be as explicit as possible what parameters they could use. I guess its more of a idk who might use this and i want to be as helpful as i can.